### ISSUE TITLE
Label is truncated when using fast update

### ISSUE THREAD

**Issue by Katherine-101** [microsoft/APSI#58]
Hi!

I have loaded SenderDB from disk to memory, and doing insert operation for a new data (item-label pair) by `SenderDB::insert_or_assign`. `label_byte_count_` in the loaded SenderDB is 7, but the label bytes of new data is 32. I found the new label is truncated to 7 bytes when the receiver get the query result.

![image](https://user-images.githubusercontent.com/106299543/233309521-9226ca75-9dfa-4e5c-8324-7dc9f2e05527.png)
![image](https://user-images.githubusercontent.com/106299543/233309254-515a4b30-6804-4414-a7cd-5229f4a092a5.png)

Is this an unresolved issue, or something wrong with my operation? 
I want to use fast update in this case (new label larger than old label in DB). My basic idea is to extend the `label_bins_` to put the extra label part, pad the other item position with zero in new bins, and then generate the polynomials.

Do you think it is workable?

**Comment by kimlaine** [comment]
Extending the labels is not supported. Why not just create a new `SenderDB` with longer labels? Label polynomial interpolation is the costliest operation anyway and it would have to be repeated since the newly extended labels need to be (re-)encrypted.

**Comment by Katherine-101** [comment]
Thank you for your reply! When I extend labels, all the labels need to be re-encrypted to get the same encrypted length. Is that right?

I have another question about cuckoo filter. I found sometimes error occurs like this when I set `felts_per_item = 8`:
![image](https://user-images.githubusercontent.com/106299543/234746516-d29cd22f-9f69-48a3-92ad-47484c812dbe.png)
Seems like the repeated felt item wasn't recognized in https://github.com/microsoft/APSI/blob/146070d95f5057977d7f10bb74a95ac97b3acb1a/sender/apsi/bin_bundle.cpp#L565
Is it the reason that some repeated felt item was kicked out of the cuckoo filter? Which parameters affect the probability of repeated points inserted in the same bin, `felts_per_item`?

**Comment by kimlaine** [comment]
Yes, when you extend labels, all labels will end up having to have the same extended length. Otherwise I'm not sure what the point of encryption would be. They need to have the same length.

As for your second question: this is very strange and perhaps there is some bug. Would you be able to share some code that reproduces this so I can look into what might be causing it?

**Comment by kimlaine** [comment]
It really might be that the cuckoo filter has some bug that causes false negatives, which should never happen no matter the parameters. The kick-out process in the cuckoo filter is just supposed to move stuff around so it's not like it should remove anything. It would be very good to know if you can reproduce this.

**Comment by kimlaine** [comment]
Closing. Let me know if you can reproduce this on v0.13.0.
